Weather App/WeathersList/viewModel/CityWeatherViewModel.swift

```diff
@@ -19,22 +19,29 @@ public protocol CityWeatherViewModel {
     /// Get weather for cities from storage.
     func setup()
     func removeWeather(index: Int)
+    /// Give weather forecast for selected index
+    /// - Parameters:
+    ///   - indexPath: IndexPath for which the weather forecast will be downloaded
+    func getForecastForCity(_ indexPath: IndexPath)
 }
 
 class CityWeatherViewModelImpl: CityWeatherViewModel {
     public var weathers: [CityWeather] = []
     public var weatherForLocation: CityWeather?
     private let citiesRepository: CitiesRepository
+    private let forecastRepository: ForecastRepositoryImpl
+
     private let storage: Storage
     private let delegate: WeathersListViewControllerDelegate
     private var cities: Cities
 
-    init(delegate: WeathersListViewControllerDelegate, storage: Storage = StorageUserDefaults(), citiesRepository: CitiesRepository = CitiesRepository()) {
+    init(delegate: WeathersListViewControllerDelegate, storage: Storage = StorageUserDefaults(), citiesRepository: CitiesRepository = CitiesRepository(), forecastRepository: ForecastRepositoryImpl = ForecastRepositoryImpl()) {
```

Comment:
I think, one repository for all requests will be enough

Reply:
👍